### PR TITLE
[IMP] l10n_in{_edi}: GSTR1 & GSTR-3B reports

### PR DESCRIPTION
**Pull request by epictete**

### CODE REVIEW

File: addons/l10n_in/data/account_tax_tag_data.xml [hunk 1]
```diff
@@ -0,0 +1,218 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <!-- GST Base tax tag -->
```
Comment (oco-odoo) [inline addons/l10n_in/data/account_tax_tag_data.xml]: Let's rather declare them in csv (waaay less lines, and the data are simple enough that we can afford that)

File: addons/l10n_in/data/account_tax_template_data.xml [hunk 2]
```diff
@@ -219,8 +8,8 @@
...
-        <field name="refund_repartition_line_ids" eval="[(5, 0, 0),             (0,0, {                 'factor_percent': 100,                 'repartition_type': 'base',             }),               (0,0, {                 'factor_percent': 100,                 'repartition_type': 'tax',                 'account_id': ref('p10055'),                 'minus_report_expression_ids': [ref('tax_report_line_cess_tag')],             }),         ]"/>
+        <field name="invoice_repartition_line_ids" eval="[(5, 0, 0),             (0,0, {                 'factor_percent': 100,                 'repartition_type': 'base',             'tag_ids': [ref('tax_tag_base_cess')]}),               (0,0, {                 'factor_percent': 100,                 'repartition_type': 'tax',                 'account_id': ref('p11235'),                 'tag_ids': [ref('tax_tag_cess')],             }),         ]"/>
```
Comment (oco-odoo) [inline addons/l10n_in/data/account_tax_template_data.xml]: Fix spacing in this file. I don't know if it was already like that, but for these, we want to use multiple lines, else it's unreadable.

File: addons/l10n_in/data/account_tax_template_data.xml [hunk 3]
```diff
@@ -1231,4 +1071,342 @@ if tax &gt; result:result=tax</field>
         <field name="sequence">0</field>
         <field name="children_tax_ids" eval="[(6, 0, [ref('sgst_purchase_2_5_rc'),ref('cgst_purchase_2_5_rc'),])]"/>
     </record>
+
+    <record id="non_gst_supplies_purchase" model="account.tax.template">
```
Comment (oco-odoo) [inline addons/l10n_in/data/account_tax_template_data.xml]: We're adding a bunch of new taxes. It's not the place to do this. Remove them from your PR, and tell JVA to do that in his branch, not here.

File: addons/l10n_in_edi/models/account_edi_format.py [hunk 4]
```diff
@@ -36,6 +36,7 @@ def _is_required_for_invoice(self, invoice):
                 "overseas",
                 "special_economic_zone",
                 "deemed_export",
+                "uin_holders",
```
Comment (oco-odoo) [inline addons/l10n_in_edi/models/account_edi_format.py:39]: What's that ?

File: addons/l10n_in_edi/models/account_edi_format.py [hunk 5]
```diff
@@ -425,19 +426,19 @@ def l10n_in_grouping_key_generator(tax_values):
             base_line = tax_values["base_line_id"]
             tax_line = tax_values["tax_line_id"]
             line_code = "other"
-            if any(tag in tax_line.tax_tag_ids for tag in self.env.ref("l10n_in.tax_report_line_cess").sudo().tag_ids):
+            if tax_line.tax_tag_ids in self.env.ref("l10n_in.tax_tag_cess"):
```
Comment (oco-odoo) [inline addons/l10n_in_edi/models/account_edi_format.py]:
```suggestion
            if self.env.ref("l10n_in.tax_tag_cess") in tax_line.tax_tag_ids:
```

I guess this makes more sense, right ?

File: addons/l10n_in_edi/models/account_edi_format.py [hunk 6]
```diff
@@ -425,19 +426,19 @@ def l10n_in_grouping_key_generator(tax_values):
...
                 if tax_line.tax_line_id.amount_type != "percent":
                     line_code = "cess_non_advol"
                 else:
                     line_code = "cess"
-            elif any(tag in tax_line.tax_tag_ids for tag in self.env.ref("l10n_in.tax_report_line_state_cess").sudo().tag_ids):
+            elif tax_line.tax_tag_ids in self.env.ref("l10n_in.tax_tag_state_cess"):
```
Comment (oco-odoo) [inline addons/l10n_in_edi/models/account_edi_format.py]:
```suggestion
            elif self.env.ref("l10n_in.tax_tag_state_cess") in tax_line.tax_tag_ids:
```

File: addons/l10n_in_edi/models/account_edi_format.py [hunk 7]
```diff
@@ -425,19 +426,19 @@ def l10n_in_grouping_key_generator(tax_values):
...
                 if tax_line.tax_line_id.amount_type != "percent":
                     line_code = "state_cess_non_advol"
                 else:
                     line_code = "state_cess"
             else:
                 for gst in ["cgst", "sgst", "igst"]:
-                    if any(tag in tax_line.tax_tag_ids for tag in self.env.ref("l10n_in.tax_report_line_%s"%(gst)).sudo().tag_ids):
+                    if tax_line.tax_tag_ids in self.env.ref("l10n_in.tax_tag_%s"%(gst)):
```
Comment (oco-odoo) [inline addons/l10n_in_edi/models/account_edi_format.py]:
```suggestion
                    if self.env.ref("l10n_in.tax_tag_%s"%(gst)) in tax_line.tax_tag_ids:
```

Comment (oco-odoo) [inline addons/l10n_in_edi/models/account_edi_format.py]: Also, rather do the ref *before* the loop.

**Review by oco-odoo — CHANGES_REQUESTED** [review]

File: addons/l10n_in_edi/models/account_edi_format.py (hunk 4 above)
Comment (epictete) [reply addons/l10n_in_edi/models/account_edi_format.py:39]: It's used in `enterprise/l10n_in_reports/data/account_financial_html_report_gstr3b.xml`

**Comment by oco-odoo** [comment]
Cherry-picked

**Pull request closed without merging**